### PR TITLE
feature/64 - support looking up facet label data from a foreign collection

### LINKED ISSUE #64: support labeling facet values from a foreign collection
like "populate" for mongo facet values -- support props for collection, foreignField and some fields to `_.pick` off the foreign record (to keep the response small), and implement a `$lookup`, `$unwind`, `$project` to pass the picked foreign props along on the facet value objects for use in (for example) contexture-react's `Facet` type `display` function https://github.com/smartprocure/contexture-react/pull/390

### PR DESCRIPTION
**Pull request by doug-patterson**
Fixes #64 

### Description
* supports specifying a couple props on your `facet` node and then getting a lookup from a foreign collection (like `populate`) for the facet values. Intent is that this be used for labeling by front-end libraries like contexture-react https://github.com/smartprocure/contexture-react/pull/390
* Taking this out of draft status as the test fails I'm seeing are present on master. Will investigate.

### CODE REVIEW

**Comment by decrapifier** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>This PR has failing tests. Please alleviate the errors and commit them

<details>
  <summary>result should output the expected query and results</summary>
  <code>expected { Object (entries) } to deeply equal { Object (entries) }</code>
</details></td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>The changelog has not been updated. Please update the changelog.</td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>The version was not updated. Please update the version.</td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>The README has not been updated. Please update the README.</td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>Could not find any browser results.</td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>We were able to automatically fix some formatting issues in this PR for you!</td>
    </tr>
  </tbody>
</table>


### Some things that were possibly fixed:

- Code that could be fixed via the --fix flag
- Formatting that could be fixed by prettier

Take a look at this commit to see what happened in detail: 6ed34891ae52eb50db62dcd45fdc226c3f96554d


And look at this wiki page to see the reasoning behind the ESLint rules: https://github.com/smartprocure/eslint-config-smartprocure/wiki/Rules-and-Why-We-Chose-Them
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 52c3e322ad772cf08b8ecbd0a06a656d33bf246d
</p>

**Comment by doug-patterson** [comment]
No way around it, I'm going to have to fix those tests even though they have nothing to do with this

**Comment by doug-patterson** [comment]
tests fixed. Ready for review. @dubiousdavid @michsa @dshishkov @Akrion someone who understands the `dateHistogram` work should check over my changes to the node type and test

File: src/example-types/facet.js [hunk 1]
```diff
@@ -2,6 +2,19 @@ let F = require('futil')
 let _ = require('lodash/fp')
 let { ObjectID } = require('mongodb')
 
+let projectStageFromLabelFields = node => {
```
Comment (doug-patterson) [inline src/example-types/facet.js]: the real changes for this issue are just the ones in this file

**Comment by doug-patterson** [comment]
@dubiousdavid @dshishkov @Akrion ready for review

**Comment by doug-patterson** [comment]
tests are failing here but not locally. closing this

**Pull request closed without merging**